From 063449df4341ee6647e17ea1714c49f938e419e1 Mon Sep 17 00:00:00 2001
From: Pino Toscano <ptoscano@redhat.com>
Date: Tue, 26 Aug 2014 14:39:28 +0200
Subject: [PATCH 053/102] daemon: move AUGEAS_ERROR to the common header

Other than for current aug_* API, it will be useful for further code
using augeas directly.

(cherry picked from commit abb0b47e3f1b8b86d2d39c2a91782ac5ed4b134e)
---
 daemon/augeas.c | 17 -----------------
 daemon/daemon.h | 17 +++++++++++++++++
 2 files changed, 17 insertions(+), 17 deletions(-)

diff --git a/daemon/augeas.c b/daemon/augeas.c
index 03a56f4..505bc2f 100644
--- a/daemon/augeas.c
+++ b/daemon/augeas.c
@@ -61,23 +61,6 @@ aug_finalize (void)
   }									\
   while (0)
 
-/* Calls reply_with_error, but includes the Augeas error details. */
-#define AUGEAS_ERROR(fs,...)                                            \
-  do {                                                                  \
-      int code = aug_error (aug);                                       \
-      if (code == AUG_ENOMEM)                                           \
-        reply_with_error (fs ": augeas out of memory", ##__VA_ARGS__);  \
-      else {                                                            \
-            const char *message = aug_error_message (aug);              \
-            const char *minor = aug_error_minor_message (aug);          \
-            const char *details = aug_error_details (aug);              \
-            reply_with_error (fs ": %s%s%s%s%s", ##__VA_ARGS__,         \
-                                message,                                \
-                                minor ? ": " : "", minor ? minor : "",  \
-                                details ? ": " : "", details ? details : ""); \
-            }                                                           \
-  } while (0)
-
 /* We need to rewrite the root path so it is based at /sysroot. */
 int
 do_aug_init (const char *root, int flags)
diff --git a/daemon/daemon.h b/daemon/daemon.h
index f2e3c7a..e6208b1 100644
--- a/daemon/daemon.h
+++ b/daemon/daemon.h
@@ -416,6 +416,23 @@ is_zero (const char *buffer, size_t size)
     }                                                   \
     while (0)
 
+/* Calls reply_with_error, but includes the Augeas error details. */
+#define AUGEAS_ERROR(fs,...)                                            \
+  do {                                                                  \
+    int code = aug_error (aug);                                         \
+    if (code == AUG_ENOMEM)                                             \
+      reply_with_error (fs ": augeas out of memory", ##__VA_ARGS__);    \
+    else {                                                              \
+      const char *message = aug_error_message (aug);                    \
+      const char *minor = aug_error_minor_message (aug);                \
+      const char *details = aug_error_details (aug);                    \
+      reply_with_error (fs ": %s%s%s%s%s", ##__VA_ARGS__,               \
+                          message,                                      \
+                          minor ? ": " : "", minor ? minor : "",        \
+                          details ? ": " : "", details ? details : ""); \
+    }                                                                   \
+  } while (0)
+
 #ifdef HAVE_ATTRIBUTE_CLEANUP
 #define CLEANUP_FREE __attribute__((cleanup(cleanup_free)))
 #define CLEANUP_FREE_STRING_LIST                        \
-- 
1.9.3

